agent/agent.go

```diff
@@ -549,6 +549,9 @@ func (a *Agent) fetchTaskInfo(ctx context.Context, tc *taskContext) (*task.Task,
 	if err != nil {
 		return nil, nil, nil, errors.Wrap(err, "getting project")
 	}
+	if project == nil {
```

Comment:
I think this case isn't possible anymore with the new logic since `GetProject` will either return on L285 or return a non-nil error on L288. Should it be removed?